### PR TITLE
Add required files for the Carvel plugin impl

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR is the first of a bunch of PRs... sorry for the upcoming reviewing work :sweat_smile: 
Here I'm just adding some required files I'll populate later on. Additionally, I'm moving some parts of the code out to different files. 

I'll be stacking the rest of the PRs onto this one.

### Benefits

An initial and very limited Carvel support in Kubeapps

### Possible drawbacks

This PR, as of today, actually reduces the functionality that had been implemented :P

### Applicable issues

- related https://github.com/kubeapps/kubeapps/issues/3221

### Additional information

Unfortunately, I haven't had yet the chance to start looking into the tests.... :disappointed: , but I wanted to start sending some PRs after having everything up and running. Next week I'll continue/start first thing on the top of my head.

Discussion about pending things: https://github.com/kubeapps/kubeapps/issues/3221#issuecomment-974545267

### CODE REVIEW

File: cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go [hunk 1]
```diff
@@ -72,473 +40,46 @@ type Server struct {
 	// clientGetter is a field so that it can be switched in tests for
 	// a fake client. NewServer() below sets this automatically with the
 	// non-test implementation.
-	clientGetter clientGetter
+	clientGetter             clientGetter
+	globalPackagingNamespace string
+	globalPackagingCluster   string
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go:45]: These surprised me? I'm sure i'll see below why they were needed.

File: cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go [hunk 2]
```diff
@@ -72,473 +40,46 @@ type Server struct {
...
-				continue
+				return nil, nil, status.Errorf(codes.FailedPrecondition, fmt.Sprintf("unable to get dynamic client : %v", err))
 			}
-			maintainerMap, found, err := unstructured.NestedStringMap(maintainerMapUnstructured)
-
-			if found && err == nil {
-				maintainers = append(maintainers, &corev1.Maintainer{
-					Name:  maintainerMap["name"],
-					Email: maintainerMap["email"],
-				})
```
Comment (absoludity) [inline cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go:450]: Just a note for next time: You can usually avoid massive diffs of moved code like this by just `git mv`ing the file which will change less and create a new one in its place once moved. Git will then (usually) track it as such so it's much easier to see if something changed in the moved code etc.

File: cmd/pinniped-proxy/Cargo.lock [hunk 3]
```diff
@@ -222,12 +222,6 @@ dependencies = [
  "winapi",
 ]
 
-[[package]]
-name = "doc-comment"
-version = "0.3.3"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "fea41bba32d969b513997752735605054bc0dfa92b4c56bf1189f2e174be7a10"
```
Comment (absoludity) [inline cmd/pinniped-proxy/Cargo.lock:229]: Why is this PR including any changes to pinniped-proxy?

File: dashboard/src/components/AppView/CustomAppView/CustomAppView.test.tsx [hunk 4]
```diff
@@ -59,6 +59,20 @@ const defaultProps = {
...
+  send: jest.fn(),
+  setRequestHeader: jest.fn(),
+  readyState: 4,
+  status: 200,
+  response: "Hello World!",
+};
+
+beforeAll((): void => {
+  jest.spyOn(window, "XMLHttpRequest").mockImplementation(() => xhrMock as XMLHttpRequest);
+});
```
Comment (absoludity) [inline dashboard/src/components/AppView/CustomAppView/CustomAppView.test.tsx:74]: Or this (which landed). Guessing you merged master into this PR but hadn't done so on the base branch that this PR targets?

**Review by absoludity — APPROVED** [review]
The PR has extra changes in it, but could be just merging our main branch here but not into your base `carvel` branch? Not sure.

File: cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go (hunk 1 above)
Comment (antgamdia) [reply cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go:45]: I'm adding these lines for defaulting to the Kubeapps cluster if no cluster is passed in the request and for entering the `kapp-controller-packaging-global` namespace if required at some point.
At this very moment, I don't know either if it's gonna be 100% required, but I added for similarity as in other plugins.

File: cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go (hunk 2 above)
Comment (antgamdia) [reply cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go:450]: Well, this particular diff I didn't know how to start with. I mean, I opted to remove the old and then add it back in the proper PR. Perhaps moving the old code to an auxiliary file and then splitting pieces of code out from there could have been another option. I agree my way is too verbose and adds unnecessary code-review complexity

File: cmd/pinniped-proxy/Cargo.lock (hunk 3 above)
Comment (antgamdia) [reply cmd/pinniped-proxy/Cargo.lock]: No, no, it is due to a base branch being outdated. I've properly rebased and targeted it

**Pull request merged**